src/main/scala/beam/utils/scenario/ScenarioLoaderHelper.scala

```diff
@@ -0,0 +1,174 @@
+package beam.utils.scenario
+
+import beam.utils.SnapCoordinateUtils
+import beam.utils.SnapCoordinateUtils.{Category, CsvFile, Error, ErrorInfo, Result, SnapLocationHelper}
+import com.typesafe.scalalogging.LazyLogging
+import org.matsim.api.core.v01.population.{Activity, Leg, Person}
+import org.matsim.api.core.v01.{Coord, Id}
+import org.matsim.core.scenario.MutableScenario
+import org.matsim.households.{Household, HouseholdsFactoryImpl}
+
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters.{collectionAsScalaIterableConverter, seqAsJavaListConverter}
+
+object ScenarioLoaderHelper extends LazyLogging {
+
+  import org.matsim.api.core.v01.population.PlanElement
+
+  private def createHouseholdWithGivenMembers(
+    household: Household,
+    members: List[Id[Person]]
+  ): Household = {
+    val householdResult = new HouseholdsFactoryImpl().createHousehold(household.getId)
+    householdResult.setIncome(household.getIncome)
+    householdResult.setVehicleIds(household.getVehicleIds)
+    householdResult.setMemberIds(members.asJava)
+    householdResult
+  }
+
+  private def updatePlanElementCoord(
+    personId: Id[Person],
+    elements: Vector[PlanElement],
+    snapLocationHelper: SnapLocationHelper
+  ): Vector[ErrorInfo] = {
+    val errors = elements.foldLeft[Vector[ErrorInfo]](Vector.empty) { (errors, element) =>
+      element match {
+        case _: Leg => errors
+        case a: Activity =>
+          val planCoord = a.getCoord
+          snapLocationHelper.computeResult(planCoord) match {
+            case Result.Succeed(_) =>
+              // note: we don't want to update coord in-place here since we might end up removing plan from the person
+              errors
+            case Result.OutOfBoundingBoxError =>
+              errors :+ ErrorInfo(
+                personId.toString,
+                Category.ScenarioPerson,
+                Error.OutOfBoundingBox,
+                planCoord.getX,
+                planCoord.getY
+              )
+            case Result.R5SplitNullError =>
+              errors :+ ErrorInfo(
+                personId.toString,
+                Category.ScenarioPerson,
+                Error.R5SplitNull,
+                planCoord.getX,
+                planCoord.getY
+              )
+          }
+      }
+    }
```

Comment:
We have a very similar code that uses `snapLocationHelper.computeResult`. We could use `HasCoord` type class to reuse this code.

Reply:
I tried to look for better to handle this as per your suggestion but I think we can't make it re-usable. Based on different logic, we do different things in those cases. Some of them accumulate errors, some modifies scenario etc. 

If you could sketch it out your idea bit more than I might able to understand your POV. Thanks.

Reply:
What I'm suggesting is somewhat similar to `beam.utils.matsim_conversion.ShapeUtils#quadTree` method.
But I'm Ok if you decide not to go with it.

Generally it looks like the following:
`SnapCoordinateUtils`:
```
import beam.utils.matsim_conversion.ShapeUtils.HasCoord
    def snapCoordinates[A: HasCoord](elements: Seq[A], isWgs: Boolean = false): Seq[(A, SnapCoordinateResult)] = {
      val hasCoord = implicitly[HasCoord[A]]
      elements.map { a =>
        val coord = hasCoord.getCoord(a)
        a -> computeResult(coord, isWgs)
      }
    }
```
You also can return some a tuple from this methos which includes other useful info (i.e. if errors are presented, in order not to go through the whole collection). You could create a `par` version of the same method for big coolections.

`ScenarioLoaderHelper`:
```
val householdsWithMembers: List[Household] = scenario.getHouseholds.getHouseholds.values().asScala.toList
    snapLocationHelper.snapCoordinates(householdsWithMembers).map{
      case (household, Right(splitCoord)) =>
        val householdId = household.getId.toString
        val attr: ObjectAttributes = ???
        attr.putAttribute(householdId, "homecoordx", splitCoord.getX)
        attr.putAttribute(householdId, "homecoordy", splitCoord.getY)
      case (household, Left(error)) =>
        household.getMemberIds.asScala.toList.foreach(personId => scenario.getPopulation.getPersons.remove(personId))
        scenario.getHouseholds.getHouseholds.remove(household.getId)
        householdErrors.append(
          ErrorInfo(
            householdId,
            Category.ScenarioHousehold,
            error,
            planCoord.getX,
            planCoord.getY
          )
        )
    }
```

Reply:
@dimaopen Thanks for detailed comment. I don't think we need this change at this point of time. Let's keep it simple for now.